### PR TITLE
Chore: Add info about K8s folder structure to README

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

Adds additional information from a discussion with @leonorfmartins to the `pkg/services/apiserver` README.

**Why do we need this feature?**

This is a quick reference for anyone that needs to navigate the current code structure.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/apiserver/README.md [hunk 1]
```diff
@@ -79,3 +79,14 @@ The Kubernetes compatible API can be accessed using existing Grafana AuthN at: [
 
 The equivalent openapi docs can be seen in [http://localhost:3000/swagger](http://localhost:3000/swagger), 
 select the relevant API from the dropdown in the upper right.
+
+## General Structure
+
```
Comment (ryantxu) [inline pkg/services/apiserver/README.md:84]:
```suggestion
The folder structure aims to follow the patterns established in standard (https://github.com/kubernetes/kubernetes)[kubernetes] projects.
```

File: pkg/services/apiserver/README.md [hunk 2]
```diff
@@ -79,3 +79,14 @@ The Kubernetes compatible API can be accessed using existing Grafana AuthN at: [
...
+* [the staging folder](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io)that is the true source of the k8s repos linked below. this is useful when tracing the source of k8s code because GitHub is able to follow where types are coming from in the Kubernetes monorepo
+* [apimachinery](https://github.com/kubernetes/apimachinery) == [pkg/apimachinery](/pkg/apimachinery) - contains types and utils that are used by both API clients and servers
+* [apiserver](https://github.com/kubernetes/apiserver) == [pkg/apiserver](/pkg/apiserver) - contains apiserver library code used for both standalone app apiservers and the one embedded in grafana. it depends on `pkg/apimachinery`
+* [pkg/services/apiserver](/pkg/services/apiserver) - this is where the embedded grafana API server background service is currently configured. it depends on `pkg/apiserver` and `pkg/apimachinery`
+* [pkg/apis](/pkg/apis) - where API resource types are defined
+* [pkg/hack/update-codegen.sh](/hack#kubernetes-hack-alert) - the script used to run [k8s codegen](https://github.com/kubernetes/code-generator/) against the types defined in `pkg/apis`
```
Comment (ryantxu) [inline pkg/services/apiserver/README.md:90]:
```suggestion
* [hack/update-codegen.sh](/hack#kubernetes-hack-alert) - the script used to run [k8s codegen](https://github.com/kubernetes/code-generator/) against the types defined in `pkg/apis`
```

File: pkg/services/apiserver/README.md (hunk 1 above)
Comment (toddtreece) [reply pkg/services/apiserver/README.md:84]: good call. the original info made more sense in the context of the conversation i had with @leonorfmartins, but without that it was very unclear. i added some additional info about how the folders are structured in [5f62507](https://github.com/grafana/grafana/pull/87508/commits/5f6250745c6a0393e919de7e8b5ae4b68911c680)

**Review by ryantxu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**